internal/handlers/common/filter.go

```diff
@@ -1435,3 +1467,53 @@ func filterFieldExprElemMatch(doc *types.Document, filterKey string, exprValue a
 
 	return filterFieldExpr(doc, filterKey, expr)
 }
+
+// filterByComparisonOperandApplicable returns true if comparison operands
+// $gt, $gte, $lt and $lte can be applied to fieldValue and exprValue.
+func filterByComparisonOperandApplicable(fieldValue, exprValue any) bool {
+	if _, ok := fieldValue.(*types.Array); ok {
+		// array is special since it could contain values
+		// which maybe the same type as operand.
+		return true
+	}
+
+	switch exprValue.(type) {
```

Comment:
Also as discussed in the call, it looks like the linter doesn't complain about types order in this case, so we need to create a new issue to investigate why the linter doesn't work.

Reply:
Fixed the order and created https://github.com/FerretDB/FerretDB/issues/1630